### DISCUSSION TITLE
Curso Fullstack .NET - Testar sem banco de dados

### DISCUSSION THREAD

**Discussion by TecnicoSLine** [balta-io/forum#583] [Tira Dúvidas] [answered]
Pessoal, estou seguindo o curso Fullstack .NET assistindo e implementando, mas, não pretendo usar EntityFramework (quero ver se implemento com Dapper), mas não queria usar banco de dados nesse momento.
É possível eu testar minha API de outra forma?
Queria tentar fazer uma implementação igual o Balta fez no curso de dominios ricos onde criava todas as regras que forem necessárias no dominio e criar testes para cada um deles.
Então... é possível eu fazer esses domínios assim e testar minha API sem banco de dados?
Me dêem uma orientação, por favor.
Obrigado.

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, Thiago, como vai?
É possível "mockar" o banco, mas talvez uma opção mais simples seja usar o EF com banco de dados em memória.
https://learn.microsoft.com/pt-br/ef/core/providers/in-memory/?tabs=dotnet-core-cli

Com ele você não vai precisar se preocupar com banco, consegue testar sua API e ainda usa EF.

Agora se não quiser usar nada de EF, aí é uma outra implementação...

**Reply by TecnicoSLine** [reply]
Fala Balta! tudo bem cara, obrigado por responder.
Então, eu não quero usar nada de EF pois aqui onde trabalho não tenho essa expectativa de utilizar, então, por mais que eu esteja estudando e fazendo projetos meus, queria continuar utilizando "ADOzão" mesmo ou até ir para um Dapper da vida.
Então... seguindo o que você disse, que outro tipo de implementação eu deveria seguir para usar ADO lá na frente e por enquanto testar minha API e faazer meus testes do dominio?
Valeu!

**Reply by andrebaltieri** [reply]
Nesse caso, você poderia utilizar o Repository Pattern, criar interfaces para seus repositórios e mockar os dados enquanto não tem banco.

Faz a carreira .NET, que nela eu ensino sobre Repository Pattern.

**Reply by TecnicoSLine** [reply]
Bom dia Balta.
Perfeito! A carreira .NET é a numero 1? Eu já até assisti um tempo atrás o curso de SQL com Dapper, mas vou repetí-lo novamente...
